frontend/src/routes/Home.svelte

```diff
@@ -1,22 +1,178 @@
 <script lang="ts">
-</script>
+  import yup from 'yup';
+  import svelteFormsLib from 'svelte-forms-lib';
+  import { _ } from 'svelte-i18n';
+  import { navigate } from 'svelte-routing';
+  import Spinner from 'svelte-tailwind-widgets/Spinner.svelte';
+  import { createConfiguration, DefaultApi, ServerConfiguration } from 'jimmi-api-client/mod.ts';
+  import config from '../config.ts';
+
+  interface IJimmiSettings {
+    domain: string;
+    room: string;
+    password: string;
+  }
+
+  const domainRegex = new RegExp(
+    /^$|^((?:(?:(?:\w[\.\-\+]?)*)\w)+)((?:(?:(?:\w[\.\-\+]?){0,62})\w)+)\.(\w{2,6})$/
```

Comment:
Like defined in [RFC 1034](https://datatracker.ietf.org/doc/html/rfc1034) every label of the domain can have up to 63 characters. So we should support also crazy long domains. The Problem is that the current magic for checking domains will break the browser for checking this lengths.

Reply:
We have seen that it is possible to generate a regular expression according to the RFC like this.
```regex
^([a-zA-Z](?:(?:([a-zA-Z]|[0-9]|-)+)?[a-zA-Z]|[0-9])?\.)*[a-zA-Z](?:(?:([a-zA-Z]|[0-9]|-)+)?[a-zA-Z]|[0-9])?$
```
This does not check the length, but whether it is a valid domain. We can use JavaScript to generate this code.
```javascript
let digit = "[0-9]";
let letter = "[a-zA-Z]";
let letDig = `${letter}|${digit}`;
let letDigHyp = `${letDig}|-`;
let ldhStr = `(${letDigHyp})+`;
let label = `${letter}(?:(?:${ldhStr})?${letDig})?`;
let subdomain = `(${label}\.)*${label}`;
let domain = subdomain;
```

Reply:
Implemented in 8499e391bf4c6a1ce5d2d142d35bda1d36f4fa11